pulsar-functions/proto/src/main/proto/Function.proto

```diff
@@ -141,6 +141,11 @@ message FunctionMetaData {
     uint64 version = 3;
     uint64 createTime = 4;
     map<int32, FunctionState> instanceStates = 5;
+    FunctionAuthenticationSpec functionAuthSpec = 6;
+}
+
+message FunctionAuthenticationSpec {
+    bytes data = 1;
```

Comment:
This should also include method, which the function runner can use to figure out which function auth provider to use.

Reply:
The function auth provider should be a cluster wide setting and not on a per function basis right?

Reply:
Please add comment to clarify the content and usage of "data" field, to summarize the below discussion.

Reply:
@merlimat i have add comments